src/plugins/discourse/references.js

```diff
@@ -59,7 +60,7 @@ export function linksToReferences(
 
   const topicRegex = new RegExp(`^(?:${topic})(?:${params})/?$`);
   const postRegex = new RegExp(`^(?:${post})(?:${params})/?$`);
-  const userRegex = new RegExp(`^(?:${server})/u/([\\w-]+)(?:${params})/?$`);
+  const userRegex = new RegExp(`^/u/([\\w-]+)(?:${params})?$`);
```

Comment:
Proposing instead of support replacing the server with `/` for all of these, we force normalize it's input URLs to be all absolute. Because in the future (unified) reference detection version, I would argue the core reference detector no longer has the context of the base URL that the URL watch fetched from.

Reply:
Great suggestion, thank you. I've implemented it, and added a bevy of new integration tests that verify that the full reference detection pipeline works on both absolute and relative references.